### DISCUSSION TITLE
Cron job server(ec2) memory keeps increasing until it shuts down

### DISCUSSION THREAD

**Discussion by saurabh-at-shvasa** [breejs/bree#185] [Q&A] [answered]
Hi,

I have a node js server running on ec2 instance with 4gb ram. In that we are using bree to run cron jobs. After restarting the server the memory goes to zero then keeps increasing as the cron jobs runs and then once it reaches max memory server f. 

Here is the main index.js file that runs

I have tried updating to bree's latest version also i.e. 9.0.1

**Comment by shadowgate15** [comment]
I don't think this is an issue with Bree but we are looking into some of our setups to see if we see a similar result. You might try running 1 job at a time to see if it is a particular job that is causing the memory creep.

**Reply by saurabh-at-shvasa** [reply]
Thanks! Yea we can try to check if any particular cron job is causing this issue but running one at a time will take some time i guess. I'll check that for sure and revert meanwhile please look if you find this issue with any other setup.

**Comment by shaunwarman** [comment]
```
// signal to parent that the job is done
if (parentPort) parentPort.postMessage('done');
else process.exit(0);
```

Are your jobs / processes properly shutting down? If you run `ps aux` for example do you see a lot of bree processes running?

**Reply by saurabh-at-shvasa** [reply]
I'll check this and will share the output thanks.

**Reply by saurabh-at-shvasa** [reply]
Hi @shadowgate15  @shaunwarman  as you can see there are total of 12 threads running for my node process and it stays 12 so i am assuming processes are not exiting? But even on running one job it says 12 so not sure are these threads related to cron jobs

<img width="697" alt="Screenshot 2022-06-27 at 11 04 11 AM" src="https://user-images.githubusercontent.com/86194282/175867215-a7ddb42d-c06f-4389-adb8-4f9d797b16f9.png">
<img width="832" alt="Screenshot 2022-06-27 at 11 04 40 AM" src="https://user-images.githubusercontent.com/86194282/175867222-ba490b3c-737d-4ca9-863a-ad49fa56374a.png">

**Comment by titanism** [comment, marked as answer]
This is not an issue with Bree, it is an issue with how you architect your jobs.
